src/shared/bootspec.c

```diff
@@ -975,6 +976,15 @@ int boot_entries_augment_from_loader(
                         .path = TAKE_PTR(p),
                         .reported_by_loader = true,
                 };
+
+                added = true;
+        }
+
+        if (added) {
+                /* If we added a new entry that the loader found, let's reasses which entries are
+                 * default/selected */
+                config->default_entry = boot_entries_select_default(config);
+                config->selected_entry = boot_entries_select_selected(config);
```

Comment:
That seems ugly, because it implies we calculate the default and selected entries twice. Shouldn't we instead postpone the calculation to do it just one at the right time?

Or maybe make `boot_entries_get_{default,selected}` caching getters that calculate the right entry on first access?